src/palace/manager/api/sirsidynix_authentication_provider.py

```diff
@@ -228,44 +230,53 @@ def remote_patron_lookup(
                 patrondata.block_reason = SirsiBlockReasons.PATRON_BLOCKED
                 return patrondata
 
-        if self.patron_status_should_block:
-            # Get patron "fines" information
-            status = self.api_patron_status_info(
-                patron_key=patrondata.permanent_id,
-                session_token=patrondata.session_token,
-            )
+        # Get patron "fines" information
+        status = self.api_patron_status_info(
+            patron_key=patrondata.permanent_id,
+            session_token=patrondata.session_token,
+        )
 
-            if not status or "fields" not in status:
-                return None
-
-            status_fields: dict = status["fields"]
-            fines = status_fields.get("estimatedFines")
-            if fines is not None:
-                # We ignore currency for now, and assume USD
-                patrondata.fines = float(fines.get("amount", 0))
-
-            # Blockable statuses
-            if status_fields.get("hasMaxDaysWithFines") or status_fields.get(
-                "hasMaxFines"
-            ):
-                patrondata.block_reason = PatronData.EXCESSIVE_FINES
-            elif status_fields.get("hasMaxLostItem"):
-                patrondata.block_reason = PatronData.TOO_MANY_LOST
-            elif status_fields.get("hasMaxOverdueDays") or status_fields.get(
-                "hasMaxOverdueItem"
-            ):
-                patrondata.block_reason = PatronData.TOO_MANY_OVERDUE
-            elif status_fields.get("hasMaxItemsCheckedOut"):
-                patrondata.block_reason = PatronData.TOO_MANY_LOANS
-            elif status_fields.get("expired"):
-                patrondata.block_reason = SirsiBlockReasons.EXPIRED
-
-            # If previously, the patron was blocked this should unset the value in the DB
-            if patrondata.block_reason is None:
-                patrondata.block_reason = PatronData.NO_VALUE
-        else:
+        if not status or "fields" not in status:
+            return None
+
+        status_fields: dict = status["fields"]
+        fines = status_fields.get("estimatedFines")
+        if fines is not None:
+            # We ignore currency for now, and assume USD
+            patrondata.fines = float(fines.get("amount", 0))
+
+        expires_date: str = status_fields.get("privilegeExpiresDate")
+        expires_date_obj = None
+
+        if expires_date:
+            expires_date_obj = dateutil.parser.parse(expires_date)
+        elif status_fields.get("expired"):
+            patrondata.block_reason = SirsiBlockReasons.EXPIRED
+        # Blockable statuses
+        if expires_date_obj and datetime.datetime.utcnow() >= expires_date_obj:
```

Comment:
This is assuming that the datetime we get is in the server local time, which I don't think we can safely assume here. Looking at the API responses, it doesn't look like it gives us any info about the timezone to expect, although we could reach out to sirsi and confirm this. 

So we don't have to deal with all this ugly timezone math, maybe we are better off just using the `expired` boolean here, rather then trying to determine if its expired ourselves.